test/unit/state/dimension.spec.js

```diff
@@ -313,5 +313,82 @@ describe('dimension', () => {
         expect(dimension.page.withMarginAndPadding).toEqual(fragment);
       });
     });
+
+    describe('calculating container dimension', () => {
```

Comment:
added a bunch of tests to this suite to ensure that margin is considered when calculating container dimension correctly